diff --git a/hw/arm/virt.c b/hw/arm/virt.c
index 219597cabd..0a5312397d 100644
--- a/hw/arm/virt.c
+++ b/hw/arm/virt.c
@@ -65,6 +65,7 @@
 #include "hw/core/platform-bus.h"
 #include "hw/core/qdev-properties.h"
 #include "hw/arm/fdt.h"
+#include "hw/misc/qemu_mbox.h"
 #include "hw/intc/arm_gic.h"
 #include "hw/intc/arm_gicv3_common.h"
 #include "hw/intc/arm_gicv3_its_common.h"
@@ -1552,6 +1553,27 @@ static void create_uart(const VirtMachineState *vms, int uart,
     g_free(nodename);
 }
 
+
+static void create_qemu_mbox(const VirtMachineState *vms)
+{
+    MachineState *ms = MACHINE(vms);
+    const hwaddr base = 0x09100000;
+    const hwaddr size = QEMU_MBOX_MMIO_SIZE;
+    const int irq = 11;
+    char *nodename;
+
+    sysbus_create_simple(TYPE_QEMU_MBOX, base, qdev_get_gpio_in(vms->gic, irq));
+
+    nodename = g_strdup_printf("/mbox@%" PRIx64, base);
+    qemu_fdt_add_subnode(ms->fdt, nodename);
+    qemu_fdt_setprop_string(ms->fdt, nodename, "compatible", "virt,mbox");
+    qemu_fdt_setprop_sized_cells(ms->fdt, nodename, "reg", 2, base, 2, size);
+    qemu_fdt_setprop_cells(ms->fdt, nodename, "interrupts",
+                           gic_fdt_irq_type_spi(vms), irq,
+                           GIC_FDT_IRQ_FLAGS_LEVEL_HI);
+    g_free(nodename);
+}
+
 static void create_rtc(const VirtMachineState *vms)
 {
     char *nodename;
@@ -3207,6 +3229,7 @@ static void machvirt_init(MachineState *machine)
     vms->highmem_ecam &= (!firmware_loaded || aarch64);
 
     create_rtc(vms);
+    create_qemu_mbox(vms);
 
     create_pcie(vms);
     create_cxl_host_reg_region(vms);
